### PR TITLE
NT-729 - Fixing issues with rejected hours in the week picker

### PR DESCRIPTION
**Pull request by eiriko2303**
So there are tree improvments/bug fixes in this PR:
- Fixed the coloring of the selected day in the dropdown. Previously the coloring just checked the current index, but now it also checks the year.

- Fixed a bug that did the following navigation: "User gets rejected hours and fix them, then he open the week-picker and click on a week for navigating to. Then he gets a popup that he must submit the rejected period. He did so, but he was sent to the first open period and not to the one he selected." With this fix it should send the user to the selected index.

- Added a subscriber for the rejectedentries inside the dropdown so that the dropdown removes the rejected tag on the period if the entries are fixed. Previously the tags in the dropdown was not updating.

### CODE REVIEW

File: src/components/submit-reopen-period/submit-reopen-period.component.ts [hunk 1]
```diff
@@ -38,11 +38,15 @@ export class SubmitReopenPeriod {
         return this.periodService.getPeriodTotalHours(this.selectedPeriod).toFixed(1);
     }
 
-    submitPeriod() {
+    submitPeriod(index?: number) {
```
Comment (reg0) [inline src/components/submit-reopen-period/submit-reopen-period.component.ts]: I'd rename this parameter somehow
maybe "forcedPeriodIndex' ?

File: src/components/top-menu-mobile/top-menu-mobile.ts [hunk 2]
```diff
@@ -100,14 +100,18 @@ export class TopMenuMobileComponent implements OnInit {
     this.displayableWeekLabel = currentweek;
   }
 
-  submitPeriod() {
+  submitPeriod(index?: number) {
```
Comment (reg0) [inline src/components/top-menu-mobile/top-menu-mobile.ts]: same here

File: src/components/week-picker-dropdown/week-picker-dropdown.component.ts [hunk 3]
```diff
@@ -39,17 +42,20 @@ export class WeekPickerDropdownComponent implements OnInit {
...
     });
-  }
-
-  ngOnInit() {
-    this.timeEntryService.getRejectedEntryDates().subscribe(rejectedDates => {
-      this.datesRejected = rejectedDates;
+    this.rejectedEntries$.subscribe(entries => {
+      let i = 0;
+      this.datesRejected = [];
+      entries.forEach((value, key, map) => {
```
Comment (reg0) [inline src/components/week-picker-dropdown/week-picker-dropdown.component.ts]:
```ts
this.datesRejected = Array.from(entries.values()).map(entry => entry.date)
``` 
maybe one liner?

File: src/pages/welcome-web/welcome-web.ts [hunk 4]
```diff
@@ -488,8 +488,8 @@ export class WelcomeWebPage {
     }
   }
 
-  submitPeriod() {
-    this.submitReopenPeriod.submitPeriod();
+  submitPeriod(index?: number) {
```
Comment (reg0) [inline src/pages/welcome-web/welcome-web.ts]: again, not meaningful param name

File: src/components/submit-reopen-period/submit-reopen-period.component.ts (hunk 1 above)
Comment (eiriko2303) [reply src/components/submit-reopen-period/submit-reopen-period.component.ts]: Agree

File: src/components/top-menu-mobile/top-menu-mobile.ts (hunk 2 above)
Comment (eiriko2303) [reply src/components/top-menu-mobile/top-menu-mobile.ts]: Agree

File: src/components/week-picker-dropdown/week-picker-dropdown.component.ts (hunk 3 above)
Comment (eiriko2303) [reply src/components/week-picker-dropdown/week-picker-dropdown.component.ts]: Yep, that is much better, thanks! 👍

File: src/pages/welcome-web/welcome-web.ts (hunk 4 above)
Comment (eiriko2303) [reply src/pages/welcome-web/welcome-web.ts]: Agree, have change them 👍

**Comment by eiriko2303** [comment]
@reg0 Should have been fixed now :)

**Pull request merged**